faunadb/objects.py

```diff
@@ -16,44 +15,46 @@ class Ref(_Expr):
   Queries that require a Ref will not work if you just pass in a string.
   """
 
-  def __init__(self, *parts):
-    """
-    Create a Ref from a string, such as ``Ref("databases/prydain")``.
-    Can also call ``Ref("databases", "prydain")`` or ``Ref(Ref("databases"), "prydain")``.
-    """
-    parts_str = "/".join(str(part) for part in parts)
+  def __init__(self, id, cls=None, db=None):
+    if id is None:
+      raise ValueError("The Ref must have an id.")
+
+    value = {"id": id}
```

Comment:
Why not store all fields separated? It seems like a better representation of what a `RefV` really is.

Reply:
I think decided to remove the db param from the red function based on feedback from java usage. There should be no need for it with the extra functions Nate added (please let me know if this is not the case)

Reply:
Ref, not red...

Reply:
I'd say to keep it in the drivers until we're able to remove the parameter from all drivers. I think removing the parameter from core won't break the driver's implementation and we're trying to keep driver's API more close to each other for maintainability.

Reply:
This is because `Ref` extends from `_Expr` and that class receives a value on its constructor

Reply:
@freels are you talking about removing the parameter from the ref expression constructor or the ref value?

Reply:
I think @freels is talking about `ref` function, not the value

Reply:
Yeah, the function. I think it is still required for the value.